tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -560,6 +560,45 @@ can_sorted_merge_append(PlannerInfo *root, CompressionInfo *info, Chunk *chunk)
 	return merge_result;
 }
 
+/*
+ * Clone the info->chunk_rel ReloptInfo and set the relid to the relid of info->compressed_rel.
+ *
+ * When we access a partially compressed chunk, we create an append path that combines the data
+ * from the compressed part with the uncompressed part. The logic in make_partition_pruneinfo()
+ * contains a check to ensure that relids are unique in an append path.
+ *
+ * To create the path that reads the uncompressed chunk data, the planner calls
+ * decompress_chunk_path_create() again. Therefore, we end up with a second path with the same
+ * RelOptInfo data structure as the real compressed path. To be compliant with the planner logic,
+ * we need to adjust the relid in one of these data structures if we want to use both paths in an
+ * append path.
+ *
+ */
+static void
+adjust_compressed_relid_in_path(PlannerInfo *root, CompressionInfo *info, Path *compressed_path,
+								Path *uncompressed_path)
+{
+	/* The compressed path and the uncompressed path share the same RelOptInfo */
+	Assert(compressed_path->parent == uncompressed_path->parent);
+
+	Oid compressed_relid = info->compressed_rel->relid;
+	Oid uncompressed_relid = info->chunk_rel->relid;
+
+	/* Ensure the relid of the uncompressed and the compressed part of the chunk are different */
+	Assert(compressed_relid != uncompressed_relid);
+
+	/* Clone the chunk_rel RelOptInfo and adjust the relid */
+	RelOptInfo *newrel_opt_info = palloc(sizeof(RelOptInfo));
+	memcpy(newrel_opt_info, info->chunk_rel, sizeof(RelOptInfo));
+	newrel_opt_info->relid = compressed_relid;
```

Comment:
This looks wrong -- RelOptInfo->relid is not an Oid, it's an index into the PlannerInfo->simple_rel_array. I think breaking this assumption can lead to weird side effects and we should probably build the proper RelOptInfos for the compressed/uncompressed tables of a partial chunk (using something like build_simple_rel).

Reply:
Looks like we use the index, but type should be Index, not Oid.

Reply:
Why don't we use the info->compressed_rel for the compressed path? I think it's meant for this.

Reply:
Good catch, the data type is changed to `Index`. 

As far as I understand our planner code, we do some modifications on `compressed_rel` (e.g., removing the metadata columns) to make it compatible with the uncompressed path in the append. I did some experiments here and was not able to modify and reuse the existing data structure without breaking the planner code.

Reply:
Append for partial chunk should be basically the same as append for table with one compressed and one uncompressed chunk, right? It doesn't look right that we can use the compressed_rel RelOptInfo for one case but not the other.

Reply:
It's a little different. If you have one fully compressed chunk `c1` and one uncompressed chunk `c2` and you perform a query you execute an `append(compressed_c1, uncompressed_c2)`.  If `c1` is only partially compressed, you execute `append(compressed_c1, uncompressed_c1, uncompressed_c2)`.  In this case, you use `compressed_c1` and `uncompressed_c1` - which need to have different `relids`.